### PR TITLE
Fixes #3 and #7 as well as misc.

### LINKED ISSUE #3: move parameters in snakefile to a config.yaml
e.g.,
`METHODS`
`METRICS`

### LINKED ISSUE #7: currently throws an error when there is more than one simulation in the simulation directory
The reason for this is these lines:
https://github.com/gwarmstrong/taxa-assign-benchmarking/blob/df0fbe6a84f20ac2e2febbbb21bd686ec90e84e3/Snakefile#L7-L9
As well as some expand's later on that essentially make the Snakefile want to look for all DT, NUM, SIM combinations

### PR DESCRIPTION
**Pull request by gwarmstrong**
This update includes a number of improvements, including:

* updated ci testing for snakemake dry-run
* support files for running snakemake (`smkjob.sh` and files it depends on)
* Fixes #3 
* Fixes #7

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+kraken_db:
```
Comment (qiyunzhu) [inline config.yaml]: Maybe differentiate `kraken` and `kraken2`? Their database formats are different.

File: ci/setup_demo_data.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+mkdir -p data/simulations/test-sim{1,2}/fake_date_sample_{0,1}/reads/
```
Comment (qiyunzhu) [inline ci/setup_demo_data.sh]: Cool! Didn't know this bash trick.

File: benchutils/metrics.py [hunk 3]
```diff
@@ -43,6 +43,8 @@ def correlation(observed, expected):
         The Pearson correlation between the series
 
     """
+    observed = np.array(observed)
```
Comment (qiyunzhu) [inline benchutils/metrics.py]: If parameters are already `np.array` (according to docstring), this conversion may be unnecessary.

File: benchutils/tests/test_metrics.py [hunk 4]
```diff
@@ -27,19 +28,19 @@ def test_correlation_basic(self):
...
         denominator = np.sqrt(denominator_sq)
         expected_correlation = numerator / denominator
-        observed_correlation = correlation(obs_profile, exp_profile)
+        observed_correlation = correlation(obs_profile, exp_profile_orig)
         self.assertAlmostEqual(observed_correlation, expected_correlation)
 
     def test_l2_norm_basic(self):
-        input_ = pd.Series([1, 0])
-        output = pd.Series([0, 0])
+        input_ = np.array([1, 0])
```
Comment (qiyunzhu) [inline benchutils/tests/test_metrics.py]: I think the tests are overall too simple. Maybe add one or two more cases per function.

File: ci/config.yaml [hunk 5]
```diff
@@ -3,3 +3,11 @@ kraken_db:
 ranks:
     # if not in a list, it will mess up
     ["genus", "species"]
+methods:
+  ["kraken2", "metaphlan2"] # "shogun"]
+metrics:
+  ['correlation', 'l2_norm', 'auprc', 'absolute_error']
```
Comment (qiyunzhu) [inline ci/config.yaml]: I wonder if `correlation` would better be `pearsonr`, as there is also `spearmanr`.

File: benchutils/metrics.py (hunk 3 above)
Comment (gwarmstrong) [reply benchutils/metrics.py]: Yeah this is leftover from debugging. Will remove.

File: config.yaml (hunk 1 above)
Comment (gwarmstrong) [reply config.yaml]: Done.

File: benchutils/tests/test_metrics.py (hunk 4 above)
Comment (gwarmstrong) [reply benchutils/tests/test_metrics.py]: I agree. Future PR.

File: ci/config.yaml (hunk 5 above)
Comment (gwarmstrong) [reply ci/config.yaml]: done.

**Pull request merged**